src/controller/CHIPDeviceControllerFactory.cpp

```diff
@@ -147,15 +147,21 @@ CHIP_ERROR DeviceControllerFactory::InitSystemState(FactoryInitParams params)
 #endif
                                                             ));
 
-    stateParams.fabricTable                                   = chip::Platform::New<FabricTable>();
     stateParams.sessionMgr                                    = chip::Platform::New<SessionManager>();
     SimpleSessionResumptionStorage * sessionResumptionStorage = chip::Platform::New<SimpleSessionResumptionStorage>();
     stateParams.sessionResumptionStorage                      = sessionResumptionStorage;
     stateParams.exchangeMgr                                   = chip::Platform::New<Messaging::ExchangeManager>();
     stateParams.messageCounterManager                         = chip::Platform::New<secure_channel::MessageCounterManager>();
     stateParams.groupDataProvider                             = params.groupDataProvider;
 
-    ReturnErrorOnFailure(stateParams.fabricTable->Init(params.fabricIndependentStorage));
+    // if no fabricTable was provided, create one and track it in stateParams for cleanup
+    FabricTable * tempFabricTable = nullptr;
+    stateParams.fabricTable       = params.fabricTable;
```

Comment:
OK.  This is still going to be broken if the controller is shut down and then restarted (at that point it will create a new, "temp" fabric table), but at least there are no memory errors anymore and fixing that can be done in a followup, I guess...

Reply:
it will require the factory to be restarted in order to restart the controller

Reply:
The point is, we support restarting controllers without restarting the factory, but if someone tries to do that it will be broken.  At the very least that limitation should be documented in the API....